aten/src/ATen/native/Normalization.cpp

```diff
@@ -19,6 +19,34 @@ static const int MIOPEN_DIM_MAX = 5;
 namespace at {
 namespace meta {
 
+static inline bool is_contiguous(const Tensor& t) {
+  return t.is_contiguous() || t.is_contiguous(at::MemoryFormat::ChannelsLast);
+}
+
+TORCH_META_FUNC(native_batch_norm)(const at::Tensor & input,
+                                   at::OptionalTensorRef weight,
+                                   at::OptionalTensorRef bias,
+                                   at::OptionalTensorRef running_mean,
+                                   at::OptionalTensorRef running_var,
+                                   bool training, double momentum, double eps) {
+
+  const bool has_running_mean = (running_mean.has_value() && running_mean->defined());
+  const bool has_running_var = (running_var.has_value() && running_var->defined());
+  TORCH_CHECK(has_running_mean == has_running_var);
+
+  int64_t n_input = input.size(1);
+  set_output(0, input.sizes(), input.options().memory_format(input.suggest_memory_format()));
+
+  auto out_stats_options = input.options();
+  if (input.scalar_type() == c10::ScalarType::BFloat16 || input.scalar_type() == c10::ScalarType::Half) {
```

Comment:
ok, so I think this logic comes from the original call to `accumulateType<...>` in the code, right? https://github.com/pytorch/pytorch/pull/67343/files#diff-e895801158bc7155ba3b61826cdd129c5e2caa5c3519171abfbe9bbe5406f52fL435.

We probably want to re-use that logic: if I'm following it correctly, it looks like the accumulate type really is the output dtype of the stats (because those stats are globals that are re-used throughout multiple calls to batch_norm).

And... I don't think there's anything backend specific about `Half/BFloat16`. I just audited it, and it looks like the [only difference happens for `float`](https://github.com/pytorch/pytorch/blob/603116a6aef8027fb363e7f081702e00cf603ff2/aten/src/ATen/AccumulateType.h#L60) (float + cpu -> float, float + cuda -> double).

So... I think maybe just re-use that logic, and add a comment calling out that we have to pass `is_cuda` into the template because cuda can have different output dtypes (and specifically mention the float->double special case for cuda for the stats). Does that sounds reasonable?